type_description_interfaces/srv/GetTypeDescription.srv

```diff
@@ -0,0 +1,43 @@
+# ROS interface type name, in PACKAGE/NAMESPACE/TYPENAME format
+string type_name
+
+# REP-2011 RIHS hash string, which looks like RIHSxx_hashvalue
+string type_version_hash
+
+# Whether to return the raw idl or msg file(s) in the response, if available
+bool include_raw_description true
+---
+# include_raw_description requested, and provided successfully
+uint8 RAW_DESCRIPTION_REQUESTED_PROVIDED = 0
+# include_raw_description disabled, so type_description_raw was left empty
+uint8 RAW_DESCRIPTION_NOT_REQUESTED_NOT_PROVIDED = 1
+# include_raw_description requested, but no raw version available (perhaps for dynamically created types)
+uint8 RAW_DESCRIPTION_REQUESTED_NOT_AVAILABLE = 2
+
+# True if the type description information is available and populated in the response
+# If false, all other fields except `reason` are considered undefined
+bool successful
+# If `successful` is false, contains a reason for failure.
+# On success, may contain a reason for missing type_description_raw.
+string reason
+
+# Contains a value from RAW_DESCRIPTION_ constants, explaining whether the
+# `type_description_raw_file_name` and `type_description_raw` fields contain valid data.
+uint8 type_description_raw_result
+
+# The idl or msg file name
+# Empty if not available, which will be noted in `reason`
+string type_description_raw_file_name
+
+# The idl or msg file, with comments and whitespace
+# The file extension and/or the contents can be used to determine the format
+# This contains all recursive raw files, each separated by a line of 80 '=' characters
+# The order will be the same as the TypeDescription -
+# which is top level interface, then referenced type descriptions in alphabetical order by type_name
+string type_description_raw
```

Comment:
After discussion in the REP-2011 meeting:

The single type description filename is not enough on its own; we actually need the file names of all types.

What we decided to do here was to instead create a structure that contains three fields:

1. The file name
2. The "encoding" (msg or IDL)
3. The description

And then in this message we have an array of that structure.

Reply:
OK - new structure added, and some language changed to call it "sources" rather than "raw descriptions" because that makes a lot more sense to me given all the discussion